src/go/configgenerator/filter_generator.go

```diff
@@ -0,0 +1,777 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configgenerator
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/options"
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util/httppattern"
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes"
+	anypb "github.com/golang/protobuf/ptypes/any"
+
+	sc "github.com/GoogleCloudPlatform/esp-v2/src/go/configinfo"
```

Comment:
File is too long to review. Recommend splitting these into separate files. For each filter, have a file that has all the filter-specific logic in that.

```
sc_filter_gen.go
jwt_filter_gen.go
```

For example, JWT and SC filters have some helper functions that are only used for that filter. Some filters have helper functions for per-route configs.

Reply:
Good point!

Done.  We have separate files for 
- service control
- jwt authn
- backend auth
- path rewrite